src/pudl_archiver/archivers/gridpathratoolkit.py

```diff
@@ -0,0 +1,133 @@
+    bucket_name = "sources.catalyst.coop"
+
+    rename_dict = {
+        "TemporalData/HourlySolar_byProject.zip": "original_solar_capacity.zip",
+        "TemporalData/HourlyWind_byProject.zip": "original_wind_capacity.zip",
+        "MonteCarlo_Inputs/temporal_data/wind/": "aggregated_wind_capacity.zip",
+        "MonteCarlo_Inputs/temporal_data/wind_syn/": "aggregated_extended_wind_capacity.zip",
+        "MonteCarlo_Inputs/temporal_data/solar/": "aggregated_solar_capacity.zip",
+        "MonteCarlo_Inputs/temporal_data/solar_syn/": "aggregated_extended_solar_capacity.zip",
+        "TemporalData/DailyWeatherData_cleaned.csv": "daily_weather.csv",
+        "SolarAggregations.csv": "solar_capacity_aggregations.csv",
+        "WindAggregations.csv": "wind_capacity_aggregations.csv",
+        "GridPath_RA_Toolkit_HowTo.pdf": "gridpathratoolkit_howto.pdf",
+        "GridPath_RA_Toolkit_Report_2022-10-12.pdf": "gridpathratoolkit_report_2022_10_12.pdf",
+        "TemporalData/readme.txt": "readme.txt",
+    }
+
+    async def get_resources(self) -> ArchiveAwaitable:
+        """Download GridPath RA Toolkit resources."""
+        bucket = storage.Client().get_bucket(self.bucket_name)
+
+        for original_file in self.rename_dict:
+            if "MonteCarlo_Inputs" not in original_file:
+                yield self.get_gcs_resource(original_file, bucket)
+            # Handle resources that need to be zipped separately
+            else:
+                yield self.get_and_zip_resources(original_file, bucket)
+
+    async def get_gcs_resource(
+        self, original_file: str, bucket: storage.Bucket
+    ) -> tuple[Path, dict]:
+        """Download GridPath RA Toolkit data files from GCS.
+
+        There are several types of files: a documentation PDF, a series of zipped files,
+        a series of CSV files,  and a series of files to download from within a
+        particular folder and zip before archiving.
+        """
+        file_name = self.rename_dict[original_file]
+        path_to_file = self.download_directory / file_name
+        blobs = bucket.list_blobs(
+            prefix=f"{self.name}/{original_file}"
+        )  # Get all blobs in folder
+
+        for i, blob in enumerate(blobs):
+            if i > 1:
+                raise AssertionError(
+                    f"More than one matching file found for {file_name}: {blob}"
+                )
+
+            # Download blob to local file
+            # We download the entire zipfile to avoid having to authenticate using
+            # a second GCS library, since GCS doesn't support fsspec file paths.
+            logger.info(f"Downloading {blob.name} to {path_to_file}")
+
+            blob.download_to_filename(path_to_file)
+
+            # The partition should be the filename without the filetype extension.
+            # E.g., solar_capacity_aggregations.csv has part: solar_capacity_aggregations
+
+        part = file_name.split(".")[0]  # Remove .csv/.zip extension
```

Comment:
Not blocking, but the `str.split("char")[idx]` pattern for munging paths stored as strings is a little hard to read and also prone to breaking when we get an unexpected filename / path. For local filesystem paths the [`pathlib.Path`](https://docs.python.org/3.12/library/pathlib.html#methods-and-properties) class has good methods for grabbing various path components, and the [`universal_pathlib`](https://github.com/fsspec/universal_pathlib) package (which I think is already installed as a dependency for `fsspec`) extends `pathlib.Path` to handle remote filesystem-like URLs.

Reply:
Good call, updated this!